src/Eagle.ts

```diff
@@ -2445,13 +2447,23 @@ export class Eagle {
             Utils.showNotification("Error", "No Node Is Selected", "warning");
         }else{
             if (tableType === 'component'){
+                if (!this.selectedNode().canHaveParameters()){
+                    Utils.showNotification("Error", "This Node Cannot Have Parameters", "warning");
+                    return
+                }
                 $("#parameterTableModalTitle").html("Component Parameter Table")
+                $("#parameterTableModalAddParameterButton").html("Add Parameter")
```

Comment:
This looks a little unusual to me. We'd normally try to do things the "knockout way" and update the html using a data-bind in the html code, rather than update it using jquery.

It would be something like:

<div data-bind="html: Eagle.parameterTableType() === Eagle.FieldType.Field ? 'Add Argument' : 'Add Parameter'">

It's about the same amount of code, so not a huge difference either way. But I guess it's more knockout-y to use the data-bind in the html code.